pkg/collector/providers/file.go

```diff
@@ -123,13 +102,54 @@ func (c *FileConfigProvider) String() string {
 	return "File Configuration Provider"
 }
 
-func collectDir(parentPath string, folder os.FileInfo) []check.Config {
+func (c *FileConfigProvider) collectEntry(entry os.FileInfo, path string, checkName string) (check.Config, bool, error) {
+	conf := check.Config{}
+	entryName := entry.Name()
+	ext := filepath.Ext(entryName)
+	isDefault := false
+
+	// skip config files that are not of type:
+	//  * check.yaml, check.yml
+	//  * check.yaml.default, check.yml.default
+	if ext == ".default" {
+		isDefault = true
+		ext = filepath.Ext(entryName[:len(entryName)-len(".default")])
+	}
+
+	if checkName == "" {
+		checkName = entryName
+		if isDefault {
+			checkName = checkName[:len(checkName)-len(".default")]
+		}
+		checkName = checkName[:len(checkName)-len(ext)]
+	}
+
+	if ext != ".yaml" && ext != ".yml" {
+		log.Debugf("Skipping file: %s", entry.Name())
+		return conf, isDefault, errors.New("Invalid config file")
```

Comment:
although the error string is not currently used, you could specify `invalid config file extension` / `invalid config file format`)